dotCMS/src/integration-test/java/com/dotcms/security/secret/SecretsStoreKeyStoreImplTest.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    /**
+     * Does storing values under the same key overrides the old value
+     */
+    @Test
+    public void Test_Storing_A_Value_Key_Name_Override() {
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+        final String key = "Any";
+        final String value1 = "value-1";
+        final String value2 = "value-2";
+        secretsStore.saveValue(key, value1.toCharArray());
+        final Optional<char[]> optionalChars1 = secretsStore.getValue(key);
+        assertTrue(optionalChars1.isPresent());
+        final String returnValue1 = new String(optionalChars1.get());
+        assertEquals("stored value is the same", value1, returnValue1);
+
+        secretsStore.saveValue(key, value2.toCharArray());
+        final Optional<char[]> optionalChars2 = secretsStore.getValue(key);
+        assertTrue(optionalChars2.isPresent());
+        final String returnValue2 = new String(optionalChars2.get());
+        assertEquals("stored value is now the second value", value2, returnValue2);
+    }
+
+    @Test
+    public void Test_404_Cache() {
+
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+
+        final String key = UUIDGenerator.generateUuid();
+        final String value = UUIDGenerator.generateUuid();
+
+        Optional<char[]> noValue = secretsStore.getValue(key);
+
+        assertFalse(noValue.isPresent());
+
+        noValue = secretsStore.getValue(key);
+
+        assertFalse(noValue.isPresent());
+
+        final char[] CACHE_404 = (char[]) CacheLocator.getCacheAdministrator()
+                .getNoThrow(key, SecretsStoreKeyStoreImpl.SECRETS_CACHE_GROUP);
+        assertNotNull(CACHE_404);
+        assertEquals(CACHE_404, SecretsStoreKeyStoreImpl.CACHE_404);
+
+        secretsStore.saveValue(key, value.toCharArray());
+        final Optional<char[]> optionalChars = secretsStore.getValue(key);
+        assertTrue(optionalChars.isPresent());
+        final String returnValue = new String(optionalChars.get());
+
+        assertEquals(value, returnValue);
+    }
+
+
+    @Test
+    public void Test_Deleting_A_Value() {
+
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+
+        final String key = UUIDGenerator.generateUuid();
+        final String value = UUIDGenerator.generateUuid();
+        secretsStore.saveValue(key, value.toCharArray());
+
+        assert (secretsStore.getValue(key).isPresent());
+        secretsStore.deleteValue(key);
+        assert (!secretsStore.getValue(key).isPresent());
+
+    }
+
+
+    @Test
+    public void Test_Empty_Value() {
+        final String uuid = UUIDGenerator.generateUuid();
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+        final Optional<char[]> val = secretsStore.getValue(uuid);
+        assert (!val.isPresent());
+    }
+
+    /**
+     * tests to make sure that the listKeys method returns newly saved keys in their list
+     */
+    @Test
+    public void Test_Value_List() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4623512)